playbooks/roles/oauth_client_setup/tasks/main.yml

```
@@ -37,5 +37,24 @@
   become_user: "{{ edxapp_user }}"
   environment: "{{ edxapp_environment }}"
   with_items: "{{ oauth_client_setup_oauth2_clients }}"
-  when: item.name != 'None'
+  when: item.name != 'None' and item.username is defined
+  tags: ["oauth_client_setup"]
+
+
+- name: create OAuth2 Clients without user
+  shell: >
+    {{ COMMON_BIN_DIR }}/python.edxapp {{ COMMON_BIN_DIR }}/manage.edxapp lms --settings={{ COMMON_EDXAPP_SETTINGS }}
+    create_oauth2_client
+    {{ item.url_root }}
+    "{{ item.url_root }}/complete/edx-oidc/"
+    confidential
+    --client_name {{ item.name }}
+    --client_id {{ item.id }}
+    --client_secret {{ item.secret }}
+    --trusted
+    --logout_uri {{ item.logout_uri | default("") }}
```

@melvinsoft I think we should keep it in the same list and use  

```suggestion
    ...
    --logout_uri {{ item.logout_uri | default("") }}
    {% if item.username %} --username {{ item.username }} {% endif %}
    ...
```

---

See https://gist.github.com/halberom/794c06598f40ccc31560#file-opt2_play-yml